### PR TITLE
Java json

### PR DESCRIPTION
**Pull request by Christian-B**
Writing a Machine out in a Java readable Json format.

### CODE REVIEW

File: pacman/utilities/file_format_converters/convert_to_java_machine.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
+from pacman.utilities import file_format_schemas
+from spinn_utilities.progress_bar import ProgressBar
+from spinn_machine.router import Router
+
+import json
+
+CHIP_HOMOGENEOUS_CORES = 18
```
Comment (alan-stokes) [inline pacman/utilities/file_format_converters/convert_to_java_machine.py]: spinnMachine (machine.py).MAX_CORES_IN_MACHINE

no point duplicating.

File: pacman/utilities/file_format_converters/convert_to_java_machine.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+
+class ConvertToJavaMachine(object):
+    """ Converter from memory machine to java machine
+    """
+
+    __slots__ = []
+
+    def __call__(self, machine, file_path):
+        """
```
Comment (alan-stokes) [inline pacman/utilities/file_format_converters/convert_to_java_machine.py]: fill in

File: pacman/utilities/file_format_converters/convert_to_java_machine.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+                    details["tags"] = list(chip.tag_ids)
+
+            if len(exceptions) > 0:
+                json_obj["chips"].append([chip.x, chip.y, details, exceptions])
+            else:
+                json_obj["chips"].append([chip.x, chip.y, details])
+        progress.update()
+
+        # dump to json file
+        with open(file_path, "w") as f:
```
Comment (alan-stokes) [inline pacman/utilities/file_format_converters/convert_to_java_machine.py]: f! f! 1 letter variable! seriously!

File: pacman/utilities/file_format_converters/convert_to_java_machine.py (hunk 1 above)
Comment (alan-stokes) [reply pacman/utilities/file_format_converters/convert_to_java_machine.py]: why its there!? not used

Comment (Christian-B) [reply pacman/utilities/file_format_converters/convert_to_java_machine.py]: Agree!

File: pacman/utilities/file_format_converters/convert_to_java_machine.py (hunk 3 above)
Comment (Christian-B) [reply pacman/utilities/file_format_converters/convert_to_java_machine.py]: https://github.com/SpiNNakerManchester/PACMAN/issues/183

File: pacman/utilities/file_format_converters/convert_to_java_machine.py (hunk 2 above)
Comment (dkfellows) [reply pacman/utilities/file_format_converters/convert_to_java_machine.py]: Fill in what? There's docs here. You might want _more_ docs… but that's just for being needlessly wordy unless you actually go to the effort to define what the actual format produced is for the docs, and that's quite a bit of work for something that almost nobody will ever read. Ever.

**Comment by Christian-B** [comment]
testing is done by 
https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/359 and 
https://github.com/SpiNNakerManchester/JavaSpiNNaker/blob/master/SpiNNaker-comms/src/test/java/uk/ac/manchester/spinnaker/transceiver/SpallocMachineTest.java

Where the second uses Data generated by the first!

**Comment by Christian-B** [comment]
Closed as now included in https://github.com/SpiNNakerManchester/PACMAN/pull/187

**Pull request closed without merging**